pkg/kubectl/cmd/cp.go

```diff
@@ -56,6 +56,9 @@ var (
 	    <file-spec> is:
 		[namespace/]pod-name:/file/path for a remote file
 		/file/path for a local file`)
+
+	// fileSpec format: [[namespace/]pod:]file/path
+	fileSpecRegexp = regexp.MustCompile(`^(?:(?:([^/:]+)/)?(?:([^/:]+):))?(.+)$`)
```

Comment:
Regexp are hard, sigh. To make it more readable can you please add more comments on top of it, something like this:
```
(?:
  (?: 
    ([^/:]+)/   // optional "namespace/" 
  )?            // non-matching group
  (?:
    ([^/:]+):   // optional "pod:"
  )             // non-matching grooup
)?
(.+)            // obligatory "file/path"
```
or even better replace my comments with the ones from the initial comment of yours from this PR.